doc/nrf/glossary.rst

```diff
@@ -430,6 +450,10 @@ Glossary
    Spinel
       A general management protocol for enabling a host device to communicate with and manage co-processors, like a network co-processor (NCP) or a radio co-processor (RCP).
 
+   Submodule
+      A different name for a *repository* managed with Git.
```

Comment:
Not all git submodules are git repositories so the terms are not equivalent. *Git submodule* is a child of another git repository, managed using `git submodule ...` command family. I'm not sure it is worth defining it here as most NCS projects don't use submodules, and *west submodules* should be rather referred to as *west projects* as we discussed. But if you would like to keep the term, the definition must be narrowed-down.